ts_src/psbt.ts

```diff
@@ -687,43 +706,47 @@ export class Psbt {
     const inputAgs: Buffer[] = [];
     const inputValues: string[] = [];
 
+    // counter for fetching blinding private key
+    let confidentialUtxoIndex = 0;
+
     // unblind inputs
     this.data.inputs.forEach((input: PsbtInput, index: number) => {
-      let prevout: any;
+      let prevout: WitnessUtxo;
       if (input.nonWitnessUtxo) {
         const prevTx = nonWitnessUtxoTxFromCache(c, input, index);
         const prevoutIndex = c.__TX.ins[index].index;
         prevout = prevTx.outs[prevoutIndex];
       } else {
-        prevout = { ...input.witnessUtxo };
+        prevout = input.witnessUtxo!;
```

Comment:
this was assigning to prevout a copy of the input's witness utxo so that future changes wouldn't have affected directly the partial transaction. I think this might be reverted. Still didn't look at this in deep.

Reply:
The `prevout` is only used to compute commitments [here](https://github.com/vulpemventures/liquidjs-lib/blob/62d68049b6779a8b873638d82c56048817659a8d/ts_src/psbt.ts#L753-L756). Why future changes would modify the partial ?

Reply:
The future PSET will change the serlization